### PR TITLE
Worker config updates to make path to jave.exe configurable

### LINKED ISSUE #3048: Remove hardcoded java version when running in Azure
Java version zulu8.23.0.3-jdk8.0.144-win_x64 is hardcoded [here](https://github.com/Azure/azure-functions-host/blob/d95653b8fda05e9a0b0c7b48196e555135a80257/src/WebJobs.Script/Rpc/Configuration/JavaWorkerProvider.cs#L85)

Need to move this to worker.config.json

### LINKED ISSUE #3187: Parse Error and Info logs from language workers with the right level
Java.exe is logging to stderr by default. Need to parse messages properly for the info/warn/error levels

### PR DESCRIPTION
**Pull request by pragnagopa**
- Added support to provide worker description profiles
- Added support to override default worker process path by specifying app setting languageWorkers:language:defaultExecutablePath

Fixes #3187, Fixes #3048

### CODE REVIEW

**Comment by pragnagopa** [comment]
Sample worker.config.json with AppServiceEnvironment profile
```
{
    "Description":{
        "Language":"java",
        "Extension":".jar",
        "DefaultExecutablePath":"java",
        "DefaultWorkerPath":"azure-functions-java-worker.jar",
        "Arguments" : ["-jar"]
    },
	"profiles": {
		"AppServiceEnvironment": {
			"DefaultExecutablePath": "../../zulu8.23.0.3-jdk8.0.144-win_x64/bin/java"
		}
	}
}
```

**Comment by pragnagopa** [comment]
Updated to Java worker.

**Comment by fabiocav** [comment]
Wondering if it would save us all some time (and avoid migration issues) if this was targeting the hosting branch instead. We're about to merge that in. What do you think?

@brettsam , thoughts?

**Comment by brettsam** [comment]
Yes! and in fact, there's some issues with worker config today in `hosting` (some values aren't getting applied like I'd expect). It'd be good to get it all reconciled at once.

File: src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs [hunk 1]
```diff
@@ -341,10 +341,14 @@ internal void StartProcess(string workerId, Process process)
                     {
                         _logger.LogWarning(e.Data);
                     }
-                    else
+                    else if (e.Data.IndexOf("error", StringComparison.OrdinalIgnoreCase) > 0)
```
Comment (fabiocav) [inline src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs]: Should this `> -1`?

File: src/WebJobs.Script/Rpc/LanguageWorkerConstants.cs [hunk 2]
```diff
@@ -24,8 +25,8 @@ public static class LanguageWorkerConstants
         public const string WorkerDescription = "Description";
         public const string WorkerDescriptionArguments = "arguments";
 
-        // Java
-        public const string AppServiceEnvJavaVersion = "zulu8.23.0.3-jdk8.0.144-win_x64";
-        public const string JavaLanguageWorkerName = "java";
+        // Profiles
+        public const string WorkerDescriptionProfiles = "profiles";
+        public const string AppServiceEnvDescription = "AppServiceEnvironment";
```
Comment (fabiocav) [inline src/WebJobs.Script/Rpc/LanguageWorkerConstants.cs]: Should this be something like `WorkerProfileNameAppServiceEnvironment` ?

**Comment by fabiocav** [comment]
@brettsam , just had a quick chat with @pragnagopa . Since we haven't changed that model for language workers in the hosting branch yet, I don't think what she is adding expands what we'd have to migrate too much, so it should be fine to go into `dev`. Thoughts?

**Comment by pragnagopa** [comment]
Right. @brettsam - I will go ahead and merge this into dev and  work with you and @fabiocav for specific changes we need to make to consume the new config model.

**Pull request merged**